### ISSUE TITLE
Sourcemaps are uploaded by gradle but are not matched against error events

### ISSUE THREAD

**Issue by andersonvom** [getsentry/sentry-react-native#1943]
### Environment

* Sentry SaaS (sentry.io)
* `react-native`: `0.63.4` (android, no hermes)
* `@sentry/react-native`: `3.2.6`
* `sentry.java.android`: `5.4.3`

### Steps to Reproduce

Follow [installation steps][1]:
```tsx
npm install --save @sentry/react-native
npx @sentry/wizard -i reactNative -p ios android

// on index.js
Sentry.init({
    dsn: "https://xxxx.ingest.sentry.io/yyyyy",
    tracesSampleRate: 1.0,
    release: 'com.example.android@0.0.4',  // tried both with and without these values
    dist: '4'                              // tried both with and without these values
});

// on component.tsx
<Pressable onPress={() => { throw new Error("Custom JS Error") }}>
  <Text>Trigger JS Error</Text>
</Pressable>
```

Build android application and upload sourcemap ([uploaded assets screenshot][2]):
```
> Task :app:bundleDataminrStagingReleaseJsAndAssets_SentryUpload_4
Processing react-native sourcemaps for Sentry upload.
> Analyzing 2 sources
> Rewriting sources
> Adding source map references
Uploading sourcemaps for release com.example.android@0.0.4+4 distribution 4
> Bundled 2 files for upload
> Uploaded release files to Sentry
> File upload complete (processing pending on server)
> Organization: <org-name>
> Project: <project-name>
> Release: com.example.android@0.0.4+4
> Dist: 4

Source Map Upload Report
  Minified Scripts
    ~/index.android.bundle (sourcemap at index.android.bundle.map)
  Source Maps
    ~/index.android.bundle.map
```

Verify sourceMappingURL: `//# sourceMappingURL=index.android.bundle.map`

Verify stackframe paths:
```json
{
  "function": "value",
  "filename": "index.android.bundle",
  "abs_path": "index.android.bundle",
  "lineno": 27,
  "colno": 720,
  "in_app": true
}
```

Try manually uploading bundle and sourcemap with:
```bash
./node_modules/@sentry/cli/bin/sentry-cli releases \
  --org <org-name> \
  --project <project-name> \
  files com.example.android@0.0.4+4 upload-sourcemaps \
  --dist 4 --strip-prefix $(pwd) \
  index.android.bundle index.android.bundle.map \
  --validate
```

### Expected Result

Stacktrace on error event should be deminified

### Actual Result

🛑  Stacktrace is still minified ([minified stacktrace][3], [tags][4] and [event link][5]).

I also noticed two strange things:
* unlike some screenshots I've seen online, [the stacktrace on sentry.io has a `java` icon][3], instead of a `JS` on the very first line (not sure if this is relevant)
* the uploaded sourcemap file has fewer entries (`sources` and `sourcesContent`) than the generated (local) sourcemap. this means that for a given stacktrace, the line/col information matches the information in the local sourcemap, but not in the uploaded one (e.g. `onPress at line 2488:2947` matches the correct source file locally, but not on the sourcemap on sentry.io)

[1]: https://docs.sentry.io/platforms/react-native/
[2]: https://user-images.githubusercontent.com/69922/145467334-342d6acf-3021-4bb0-8983-e8ba03b96372.png
[3]: https://user-images.githubusercontent.com/69922/145468410-a5a98ad7-5331-4a7b-b6d4-e7839a53a742.png
[4]: https://user-images.githubusercontent.com/69922/145468574-1b4da32a-d1bb-47e4-bbf2-75ec6ec2896e.png
[5]: https://sentry.io/share/issue/be94351f3eee4d598c141f0a441df787/

**Comment by marandaneto** [comment]
@andersonvom please provide a minimal reproducible example.

> Try manually uploading bundle and sourcemap with:

you don't need that if you are not using custom release/dist

also the event link is broken, https://sentry.io/share/issue/9b53f38e21684dfcb4546daf94631598/, can you double check? thanks

**Comment by andersonvom** [comment]
Hi @marandaneto! I think I accidentally deleted the previous event. I've updated the description with a working event link: https://sentry.io/share/issue/be94351f3eee4d598c141f0a441df787/ . For extra context, I just tried to manually upload the sourcemap because even after verifying the bundle+sourcemap had been uploaded by gradle, the stacktraces were still minified, and I wanted to verify whether the same thing would happen with a manual upload (which it still did).

I'll try to put together a smaller environment next week (I'm currently OOO), but in the meantime please let me know if there's something that was clearly missed.

For what it's worth, another issue I did see, is that I had to wrap the initialization code (i.e. `Sentry.init(...)`) in a `try`/`catch` block, otherwise the app fails to start due to:

```
ERROR    TypeError: global.document.addEventListener is not a function. (In 'global.document.addEventListener('click', globalDOMEventHandler, false)', 'global.document.addEventListener' is undefined)
ERROR    Invariant Violation: Module AppRegistry is not a registered callable module (calling runApplication)
ERROR    Invariant Violation: Module AppRegistry is not a registered callable module (calling runApplication)
```

I wonder if there's extra code that would have run after the error and that's what's causing the the trace issue.

**Comment by marandaneto** [comment]
@andersonvom the problem is probably this error then, since the Android SDK is capturing the error and not the JS layer that's why symbolication does not work.
See https://github.com/getsentry/sentry-react-native/issues/785 and https://github.com/getsentry/sentry-react-native/issues/1203 maybe
you don't need to wrap the SDK in a try/catch block, if you solve this issue, the symbolication should work just fine

**Comment by andersonvom** [comment]
Hey @marandaneto, thanks for the pointers! Yup, it turns out the problem was a clash with [`okta-react-native`][1]. I guess they write something to `global.document` 🤔.

For others who might see similar issues in the future, the following was needed in order to allow the full sentry initialization to complete without errors:

```ts
global.window.location = { protocol: '' };
global.window.navigator.userAgent = '';

Sentry.init({
  // ...
  integrations: [
    new Sentry.BrowserIntegrations.Breadcrumbs({ dom: false }),
  ],
});
```


[1]: https://github.com/okta/okta-react-native
